flyway-core/src/main/java/org/flywaydb/core/internal/dbsupport/DbSupport.java

```diff
@@ -144,6 +143,30 @@ public void restoreCurrentSchema() {
     protected abstract void doChangeCurrentSchemaTo(String schema) throws SQLException;
 
     /**
+     * Retrieves the current user.
+     *
+     * @return The current user for this connection.
+     */
+    public String getCurrentUserName() {
```

Comment:
My point is that this is probably just an Oracle-specific thing. Since Oracle is really the only database blurring the line between schema and user and therefore it probably doesn't belong in this general interface and should be kept to the Oracle-specific part.